src/Input/Input.jsx

```diff
@@ -30,16 +32,20 @@ export default function Input({
             placeholder={placeholder}
             className={padding}
             ref={rel}
-            onChange={
-              validateOnChange
-                ? e => onChange(formData(e.target.form || {}), e.target.id)
-                : e => cleanField(e.target.id)
-            }
-            onBlur={
-              validateOnBlur
-                ? e => onBlur(formData(e.target.form || {}), e.target.id)
-                : null
-            }
+            onChange={e => {
+              if (validateOnChange) {
+                validate(formData(e.target.form || {}), e.target.id);
```

Comment:
hey jo! 
`e => onChange(formData(e.target.form || {}), e.target.id)`
this will return a function but currently there is no function being returned. How do we ensure that only 
`validate(formData(e.target.form || {}), e.target.id);`, for instance, gets triggered and not both this and `onChange(e);`
?

Reply:
>this will return a function but currently there is no function being returned. How do we ensure that only

what do you mean? event handlers should not return a function

We _do_ call both validate and onChange, I can imagine scenarios where you want to know the value but also want the validation to still function?

Reply:
i see